ktor-client/ktor-client-core/common/src/io/ktor/client/features/tracing/Tracer.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.features.tracing
+
+import io.ktor.client.request.*
+
+/**
+ * Tracer interface invoked at crucial points of the request processing to handle important events such as a start of
+ * the request processing, a receiving of response headers, a receiving of data and so on. Implementations of this
+ * interface are responsible for saving and presenting these events.
+ */
+interface Tracer {
```

Comment:
consider extracting tracer feature to the separate module

Reply:
Done.